DependencyInjection/Compiler/SecurityMiddlewarePass.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace League\Tactician\Bundle\DependencyInjection\Compiler;
+
+use League\Tactician\Bundle\Middleware\SecurityMiddleware;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * This compiler pass registers security middleware if possible
+ */
+class SecurityMiddlewarePass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (false === $container->hasDefinition('security.authorization_checker')) {
+            return;
+        }
+
+        $container->setDefinition(
+            SecurityMiddleware::SERVICE_ID,
```

Comment:
I don't think it's the responsibility of the middleware (not just `SecurityMiddleware` but also the others) to know how there are known in the DIC.